schema/resourcedefinition.proto

```diff
@@ -0,0 +1,39 @@
+// resourcedefinition contains the
+// schema of the resource definition.
+// regenerate with
+//
+// protoc ./aepc/schema/resourcedefinition.proto --go_opt paths=source_relative\
+//   --go_out=.
+syntax = "proto3";
+option go_package = "/schema";
+
+message Service {
+  string name = 1;
+  repeated Resource resources = 2;
+}
+
+message Resource {
```

Comment:
- parent

Reply:
Or parent**s**, plural...

Other things you can say about a resource, from the perspective of Google's AIPs:

* Is it a singleton?
* Is it versioned?
* Is it implicit?  (that is, does it have a schema at all?)
* What is the format/pattern of its ID?
* Which standard fields does it have?

There's also some stuff around naming...depending on what you're generating, you may need both the singular name and the plural name.  You might even need some hints about whether and how to abbreviate the collection name (or equivalent path segment for singletons) when it comes to resource names that start with the name of an ancestor resources.  For example, if you have DataStore and DataStoreEntry, you might want a URI like `data-stores/123/entries/456` rather than `data-stores/123/data-store-entries/456`.  It's possible this can be fully automated, but there might be edge cases that make fully specifying it tough, vs. asking people to specify it and then validating it.

Reply:
I think we should fully automate as much as we can - I do accept for adoption purposes we may need to budge on some of the conventions, but I'd like to debate each one a little bit before we add exceptions.

as we do, I think we need a "strict" mode - one where no conventions are broken, and brand new APIs can adhere to them.

We probably want to discuss each of these in individual issues, but:

> Is it a singleton?

I've filed #3 to discuss futher.

> Is it versioned?

Discuss that in #2?

> Is it implicit? (that is, does it have a schema at all?)

Filed #4.

> What is the format/pattern of its ID?

You can figure that out by looking at the output: the resource name is `{singular}/{id}`. We should update the AIPs but I think there should really only be one way to declare a pattern.

> Which standard fields does it have?

This is probably an iterative question? the output proto has the standard fields added. You won't find them in the resource definition itself, since they're known fields that all resources have.

> There's also some stuff around naming...depending on what you're generating, you may need both the singular name and the plural name. 

I dont like the plural name being used in code-generated paths like resource name, but I guess we need it for documentation. I'll add it.